centreon/www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -568,3 +386,245 @@ function createTimePeriodsExceptions(array $params): void
+
+        return $timeperiodId;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
+
+        return null;
+    }
+}
+
+/**
+ * Make the API request to create a new timeeperiod and return the new ID.
+ *
+ * @param array $formData
+ * @param string $basePath
+ *
+ * @throws LogicException
+ * @throws Exception
+ *
+ * @return int
+ */
+function insertTimeperiodByApi(array $formData, string $basePath): int
+{
+    $kernel = Kernel::createForWeb();
+    /** @var Router $router */
+    $router = $kernel->getContainer()->get(Router::class);
+    $client = new CurlHttpClient();
+
+    $payload = getPayloadForTimePeriod($formData);
+    $url = $router->generate(
+        'AddTimePeriod',
+        $basePath ? ['base_uri' => $basePath] : [],
+        UrlGeneratorInterface::ABSOLUTE_URL,
+    );
+
+    $headers = [
+        'Content-Type' => 'application/json',
+        'Cookie' => 'PHPSESSID=' . $_COOKIE['PHPSESSID'],
+    ];
+    $response = $client->request(
+        'POST',
+        $url,
+        [
+            'headers' => $headers,
+            'body' => json_encode($payload),
+        ],
+    );
+
+    if ($response->getStatusCode() !== 201) {
+        $content = json_decode($response->getContent(false));
+
+        throw new Exception($content->message ?? 'Unexpected return status');
+    }
+
+    $data = $response->toArray();
+
+    /** @var array{id:int} $data */
+    return $data['id'];
+}
+
+/**
+ * Update a timeperiod.
+ *
+ * @param mixed $tp_id
+ *
+ * @return bool
+ */
+function updateTimeperiodInAPI($tp_id = null): bool
+{
+    if (! $tp_id) {
+        return true;
+    }
+
+    global $form, $basePath;
+
+    $formData = $form->getSubmitValues();
+
+    try {
+        updateTimeperiodByApi($formData, $basePath);
+
+        return true;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
+
+        return false;
+    }
+}
+
+/**
+ * Make the API request to update a timeeperiod .
+ * @param array $formData
+ * @param string $basePath
+ *
+ * @throws LogicException
+ * @throws Exception
+ *
+ * @return void
+ */
+function updateTimeperiodByApi(array $formData, string $basePath): void
+{
+    $kernel = Kernel::createForWeb();
+    /** @var Router $router */
+    $router = $kernel->getContainer()->get(Router::class);
+    $client = new CurlHttpClient();
+
+    $payload = getPayloadForTimePeriod($formData);
+    $url = $router->generate(
+        'UpdateTimePeriod',
+        $basePath ? ['base_uri' => $basePath, 'id' => $formData['tp_id']] : [],
+        UrlGeneratorInterface::ABSOLUTE_URL,
+    );
+
+    $headers = [
+        'Content-Type' => 'application/json',
+        'Cookie' => 'PHPSESSID=' . $_COOKIE['PHPSESSID'],
+    ];
+    $response = $client->request(
+        'PUT',
+        $url,
+        [
+            'headers' => $headers,
+            'body' => json_encode($payload),
```

Comment:
json_encode can return false on failure, you can use [JSON_THROW_ON_ERROR] (https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag to improve error handling ;)